src/main/java/rusbik/database/RusbikLogger.java

```diff
@@ -0,0 +1,38 @@
+package rusbik.database;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RusbikLogger extends Thread{
+    private volatile List<RusbikBlockAccionPerformLog> blockAccitionPerformLogs = new ArrayList<>();
+
+    public RusbikLogger() {
+        this.start();
+    }
+    
+    /**
+     * Add log to the log list.
+     * @param log with the information of the action carried out on the block
+     */
+    public synchronized void addBlockAccionPerformLog( RusbikBlockAccionPerformLog log){
+         this.blockAccitionPerformLogs.add(log);
+     }
+     /**
+      * Thread of writing logs in the database.
+      */
+    @Override
+     public void run(){
+         while(true){
+             if(!blockAccitionPerformLogs.isEmpty()){
+                try {
+                    RusbikDatabase.blockLogging(this.blockAccitionPerformLogs.get(0));
```

Comment:
Es mejor esto o hacer un bucle for?, puede que haya más de 1 elemento en la array.

Reply:
Cuando el server esta en funcionamiento el thread minecraft y el thread logger acceden de forma concurrente al ArrayList blockAccitionPerformLogs por lo que para evitar problemas de concurrencia el thread minecraft escribe al final y thread logger lee y borra del principio.

Si pongo un for no se puede garantizar que el arreglo tenga tamaño constante y dará un error de concurrencia.

Cuando se detiene el servidor de minecraft en el método de cierre verifico que no queden logs pendientes de escribir, y si es así los escribo desde un forEach.

Reply:
entiendo, habría que poner un lock, bueno da igual, al cerrar el server estamos seguros de que termina de escribir toda la array con el forEach así que done